sdk/identity/azure-identity-broker/azure/identity/broker/_utils.py

```diff
@@ -0,0 +1,109 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+from typing import List, Optional
+import os
+import functools
+import logging
+from contextvars import ContextVar
+from azure.core.exceptions import ClientAuthenticationError
+
+
+class EnvironmentVariables:
+    AZURE_CLIENT_ID = "AZURE_CLIENT_ID"
+    AZURE_CLIENT_SECRET = "AZURE_CLIENT_SECRET"
+    AZURE_TENANT_ID = "AZURE_TENANT_ID"
+    CLIENT_SECRET_VARS = (AZURE_CLIENT_ID, AZURE_CLIENT_SECRET, AZURE_TENANT_ID)
+
+    AZURE_CLIENT_CERTIFICATE_PATH = "AZURE_CLIENT_CERTIFICATE_PATH"
+    AZURE_CLIENT_CERTIFICATE_PASSWORD = "AZURE_CLIENT_CERTIFICATE_PASSWORD"
+    CERT_VARS = (AZURE_CLIENT_ID, AZURE_CLIENT_CERTIFICATE_PATH, AZURE_TENANT_ID)
+
+    AZURE_USERNAME = "AZURE_USERNAME"
+    AZURE_PASSWORD = "AZURE_PASSWORD"
+    USERNAME_PASSWORD_VARS = (AZURE_CLIENT_ID, AZURE_USERNAME, AZURE_PASSWORD)
+
+    AZURE_POD_IDENTITY_AUTHORITY_HOST = "AZURE_POD_IDENTITY_AUTHORITY_HOST"
+    IDENTITY_ENDPOINT = "IDENTITY_ENDPOINT"
+    IDENTITY_HEADER = "IDENTITY_HEADER"
+    IDENTITY_SERVER_THUMBPRINT = "IDENTITY_SERVER_THUMBPRINT"
+    IMDS_ENDPOINT = "IMDS_ENDPOINT"
+    MSI_ENDPOINT = "MSI_ENDPOINT"
+    MSI_SECRET = "MSI_SECRET"
+
+    AZURE_AUTHORITY_HOST = "AZURE_AUTHORITY_HOST"
+    AZURE_IDENTITY_DISABLE_MULTITENANTAUTH = "AZURE_IDENTITY_DISABLE_MULTITENANTAUTH"
+    AZURE_REGIONAL_AUTHORITY_NAME = "AZURE_REGIONAL_AUTHORITY_NAME"
+
+    AZURE_FEDERATED_TOKEN_FILE = "AZURE_FEDERATED_TOKEN_FILE"
+    WORKLOAD_IDENTITY_VARS = (AZURE_AUTHORITY_HOST, AZURE_TENANT_ID, AZURE_FEDERATED_TOKEN_FILE)
+
+
+within_dac = ContextVar("within_dac", default=False)
```

Comment:
I still think we should make an exception to the no internal classes being imported for this specific case. As it is now, this will never get set, since DAC will set the `within_dac` instance from the base identity class.